File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& m) {
        int n = m.size() - 1;
        
        for(int i = 0; i < m.size(); i++) {
            for(int j = 0; j < i; j++) {
                swap(m[i][j], m[j][i]);
            }
        }
        
        for(int i = 0; i < n + 1; i++)
            reverse(begin(m[i]), end(m[i]));
    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        # Tranpose and reverse each row
        # Transpose
        for r in range(len(matrix)):
            for c in range(r, len(matrix)):
                matrix[r][c], matrix[c][r] = matrix[c][r], matrix[r][c]
        
        for r in matrix:
            r.reverse()